### PR TITLE
lib/popcorn: introduce PopTracker

### PR DESCRIPTION
**Pull request by Morriar**
## Popcorn web tracker

Easy and ready to use web tracker you can apply to your popcorn application.

The only thing you have to do is to use the tracker in your app routes:

~~~nit
var config = new AppConfig
var app = new App
app.use("/", new HomeHandler)
app.use("/products", new ProductsHandler)
app.use("customers/", new CustomersHandler)

app.use_after("/*", new PopTracker(config)) # tracker listens to /*
~~~

You can also use multiple tracker at once on different route.
All the data will be aggregated for you.

~~~nit
app.use_after("/api/*", new PopTracker(config))
app.use_after("/admin/*", new PopTracker(config))
~~~

To retrieve your tracker data use the `PopTrackerAPI` which serves the tracker
data in JSON format.

~~~nit
app.use("/api/tracker_data", new PopTrackerAPI(config))
~~~

Here some demo of what can be done when using the data:

image
image
image
image

More here: http://nitweb.moz-code.org/traffic

### CODE REVIEW

**Comment by privat** [comment]
test this please

File: lib/popcorn/pop_tracker.nit [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	# Logs collection
+	var tracker_logs = new TrackerRepo(db.collection("tracker_logs"))
+
+	# Tracker handler
+	var tracker = new PopTracker(self)
+end
+
+# JSON API of the PopTracker
+#
+# Serves the collected data in JSON format.
```
Comment (jcbrinfo) [inline lib/popcorn/pop_tracker.nit]: The API should be documented so the users don’t have to guess how to use it (or dig in the source code).

File: lib/popcorn/pop_tracker.nit [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+	end
+end
+
+# Group and count entries by browser
+class PopTrackerBrowsers
+	super TrackerHandler
+
+	# MongoMatch query used for each browser key
+	#
+	# Because parsing user agent string is a pain in the nit, we go lazy on this
```
Comment (privat) [inline lib/popcorn/pop_tracker.nit]: !

**Comment by privat** [comment]
+1
What about the diagrams?

**Comment by Morriar** [comment]
The diagrams are written in JS/d3js, not sure how I can move this to the lib :/

**Comment by Morriar** [comment]
@ppepos maybe with a template engine?

**Pull request merged**